warehouse/integrations/snowflake/snowflake_test.go

```diff
@@ -1258,6 +1258,92 @@ func TestIntegration(t *testing.T) {
 		require.NoError(t, rows.Err())
 		require.Empty(t, ids2, "no more rows left")
 	})
+
+	t.Run("Add Columns", func(t *testing.T) {
+		ctx := context.Background()
+		namespace := whth.RandSchema(destType)
+
+		credentialsJSON, err := jsonrs.Marshal(sqlconnectconfig.Snowflake{
+			Account:   credentials.Account,
+			User:      credentials.User,
+			Role:      credentials.Role,
+			Password:  credentials.Password,
+			DBName:    credentials.Database,
+			Warehouse: credentials.Warehouse,
+		})
+		require.NoError(t, err)
+
+		sqlConnectDB, err := sqlconnect.NewDB("snowflake", credentialsJSON)
+		require.NoError(t, err)
+
+		db := sqlConnectDB.SqlDB()
+		require.NoError(t, db.Ping())
+		t.Cleanup(func() { _ = db.Close() })
+		t.Cleanup(func() {
+			dropSchema(t, db, namespace)
+		})
+
+		sf := snowflake.New(config.New(), logger.NOP, stats.NOP)
+		sf.DB = sqlquerywrapper.New(db)
+		sf.Namespace = namespace
+
+		_, err = sf.DB.ExecContext(ctx, fmt.Sprintf(`CREATE SCHEMA IF NOT EXISTS %s`, namespace))
+		require.NoError(t, err, "should create schema")
+
+		tableName := "TEST_TABLE"
+		_, err = sf.DB.ExecContext(ctx, fmt.Sprintf(`CREATE TABLE %s.%q (col1 VARCHAR)`, namespace, tableName))
+		require.NoError(t, err, "should create table")
+
+		testCases := []struct {
+			name        string
+			columnsInfo []whutils.ColumnInfo
+		}{
+			{
+				name: "add single new column",
+				columnsInfo: []whutils.ColumnInfo{
+					{Name: "col2", Type: "string"},
+				},
+			},
+			{
+				name: "add multiple new columns",
+				columnsInfo: []whutils.ColumnInfo{
+					{Name: "col3", Type: "int"},
+					{Name: "col4", Type: "float"},
+				},
+			},
+			{
+				name: "add single existing column",
+				columnsInfo: []whutils.ColumnInfo{
+					{Name: "col2", Type: "string"},
+				},
+			},
+			{
+				name: "add multiple new + existing columns",
+				columnsInfo: []whutils.ColumnInfo{
+					{Name: "col2", Type: "string"},
+					{Name: "col3", Type: "int"},
+					{Name: "col4", Type: "float"},
```

Comment:
What is the new column here? All three columns are already present?

Reply:
Can we add another test with col3, col4, col5 and col6?

Reply:
These were not supposed to be independent tests but rather sequential. Now that you have pointed it out, I shouldn't be using `t.Run` here. Refactored the code with a simple for loop